File: dataset/output-bk/MISTRAL_codestral-latest/Task13_MISTRAL_codestral-latest.java
// Java
public class Main {
    public static boolean isValidWalk(char[] walk) {
        if (walk.length != 10) return false;
        int x = 0, y = 0;
        for (char dir : walk) {
            switch (dir) {
                case 'n': y++; break;
                case 's': y--; break;
                case 'e': x++; break;
                case 'w': x--; break;
            }
        }
        return x == 0 && y == 0;
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task13_MISTRAL_codestral-latest.cpp
#include <vector>

bool isValidWalk(std::vector<char> walk) {
    if (walk.size() != 10) {
        return false;
    }
    int x = 0, y = 0;
    for (char direction : walk) {
        if (direction == 'n') {
            y++;
        } else if (direction == 's') {
            y--;
        } else if (direction == 'e') {
            x++;
        } else if (direction == 'w') {
            x--;
        }
    }
    return x == 0 && y == 0;
}